---
title: "GeneticDiversity"
author: "Akira Hirao"
date: "`r Sys.Date()`"
output:
  md_document:
    toc: true
    variant: markdown_github
  html_document:
    toc: true
---

# Genetic diversity assessmment

## Loading packages

```{r message = FALSE, warning = FALSE, echo = TRUE}
# Loading packages
library(stringr)
library(hierfstat)
library(adegenet)
library(poppr)
library(strataG) #for performing M-ratio
```

## Loading data set

```{r message = FALSE, warning = FALSE, echo = TRUE}
#initializing
rm(list = ls())

# Loading data set
nameko.raw <- read.csv("MLG_Pmicro_124samples.csv",header=T)

# define locus name
locus.names <- unique(str_sub(colnames(nameko.raw)[-c(1:4)],end=-2))
no.locus <- length(locus.names)

# Convert to 6-digit-numeric coded-genotype
nameko.6_digit.genotype.raw = data.frame()
nameko.6_digit.genotype.raw = data.frame(nameko.raw[,c(1:4)])
for (i in 1:no.locus){
  allele.A.position = 3 + i*2
  allele.B.position = 4 + i*2
  target.loci = locus.names[i]
  nameko.6_digit.genotype.raw = data.frame(nameko.6_digit.genotype.raw, target.loci = paste(formatC(nameko.raw[,allele.A.position],width=3, flag="0"),formatC(nameko.raw[,allele.B.position],width=3, flag="0"),sep=""))
}
colnames(nameko.6_digit.genotype.raw)[-c(1:4)] <- locus.names

#filtering out the sample "K23" because of missing alleles as expressed "NA"
nameko.6_digit.genotype  <- na.omit(nameko.6_digit.genotype.raw)

# Convert to genind
nameko.SSR.genind <- df2genind(nameko.6_digit.genotype[,-c(1:4)],ploidy=2,ncode=3,ind.name=nameko.6_digit.genotype$ID,pop=nameko.6_digit.genotype$Category)

# Convert to genind
nameko.SSR.subpop.genind <- df2genind(nameko.6_digit.genotype[,-c(1:4)],ploidy=2,ncode=3,ind.name=nameko.6_digit.genotype$ID,pop=nameko.6_digit.genotype$Source)

strata(nameko.SSR.genind) <- data.frame(nameko.6_digit.genotype[,c(2:3)])

# Convert to genclone
nameko.SSR.genclone <- as.genclone(nameko.SSR.genind)
nameko.SSR.subpop.genclone <- as.genclone(nameko.SSR.subpop.genind)

#Repeat motif of each of the 14 SSR loci
pinfreps <- c(3,2,3,2,2,2,2,3,3,3,2,2,3,2)

#Clone correction
nameko.SSR.MLG.genind <- clonecorrect(nameko.SSR.genind)
nameko.SSR.MLG.subpop.genind <- clonecorrect(nameko.SSR.subpop.genind)
no.MLG.samples <- length(indNames(nameko.SSR.MLG.genind))

nameko.SSR.MLG.three.categories.genind <- nameko.SSR.MLG.genind
subpop.MLG.list <- nameko.SSR.MLG.subpop.genind@pop
MLG.ID.cultivar.sawdust <- which(subpop.MLG.list %in% "Cultivar.sawdust.JPN")
three.categories.MLG.list <- as.character(nameko.SSR.MLG.genind@pop)
three.categories.MLG.list[MLG.ID.cultivar.sawdust] <- "Cultivar.sawdust.JPN"
three.categories.MLG.list[three.categories.MLG.list=="Cultivar"] <- "Cultivar.others"
three.categories.factor.list <- sort(unique(three.categories.MLG.list))
three.categories.MLG.list <- factor(three.categories.MLG.list,levels=three.categories.factor.list)
nameko.SSR.MLG.three.categories.genind@pop <- three.categories.MLG.list

nameko.SSR.MLG.overall.genind <- nameko.SSR.MLG.genind
overall.MLG.list <- rep("overall",no.MLG.samples)
overall.MLG.list <- factor(overall.MLG.list)
nameko.SSR.MLG.overall.genind@pop <- overall.MLG.list

```

## Summary statistics

```{r message = FALSE, warning = FALSE, echo = TRUE}
# Defining SE function
std_mean <- function(x) sd(x)/sqrt(length(x))

# Summary
summary.out <- summary(nameko.SSR.MLG.three.categories.genind)

# Number of alleles
A.out <- allele.count(nameko.SSR.MLG.three.categories.genind)
A.overall <- nAll(nameko.SSR.MLG.overall.genind,onlyObserved = TRUE)
A.cultivar.sawdust <- nAll(nameko.SSR.MLG.three.categories.genind[pop="Cultivar.sawdust.JPN"],onlyObserved = TRUE)
A.cultivar.others <- nAll(nameko.SSR.MLG.three.categories.genind[pop="Cultivar.others"],onlyObserved = TRUE)
A.wild <- nAll(nameko.SSR.MLG.three.categories.genind[pop="Wild"],onlyObserved = TRUE)

# Percentage of polymophic loci
P.overall <- sum(A.overall > 1)/length(A.overall)
P.cultivar.sawdust <- sum(A.cultivar.sawdust > 1)/length(A.cultivar.sawdust)
P.cultivar.others <- sum(A.cultivar.others > 1)/length(A.cultivar.others)
P.wild <- sum(A.wild > 1)/length(A.wild)

cat("P: cultivar.sawdust.JPN\n")
print(P.cultivar.sawdust)
cat("\n")

cat("P: cultivar.others\n")
print(P.cultivar.others)
cat("\n")

cat("P: wild\n")
print(P.wild)
cat("\n")

cat("P: overall\n")
print(P.overall)
cat("\n")

cat("Number of alleles: cultivar.sawdust.JPN\n")
print(mean(A.cultivar.sawdust))
print(std_mean(A.cultivar.sawdust))
cat("\n")

cat("Number of alleles: cultivar.others\n")
print(mean(A.cultivar.others))
print(std_mean(A.cultivar.others))
cat("\n")

cat("Number of alleles: wild\n")
print(mean(A.wild))
print(std_mean(A.wild))
cat("\n")

cat("Number of alleles: overall\n")
print(mean(A.overall))
print(std_mean(A.overall))
cat("\n")

# Allelic richness
AR.out <- hierfstat::allelic.richness(nameko.SSR.MLG.three.categories.genind, min.n = 10)
AR.overall.out <- hierfstat::allelic.richness(nameko.SSR.MLG.overall.genind, min.n = 10)

cat("Allelic richness: mean\n")
print(apply(AR.out$Ar,2,mean))
cat("Allelic richness: SE\n")
print(apply(AR.out$Ar,2,std_mean))
cat("\n")

cat("Allelic richness across overall: mean\n")
print(apply(AR.overall.out$Ar,2,mean)[[1]])
cat("Allelic richness across overall: SE\n")
print(apply(AR.overall.out$Ar,2,std_mean)[[1]])
cat("\n")

# Observed heterozygosity
Ho.out <- Ho(nameko.SSR.MLG.three.categories.genind)
cat("Observed heterozygosity\n")
print(Ho.out)
cat("Observed heterozygosity across overall\n")
print(Ho(nameko.SSR.MLG.overall.genind)[[1]])
cat("\n")


# Expected heterozygosity
Hs.out <- Hs(nameko.SSR.MLG.three.categories.genind)
cat("Expected heterozygosity\n")
print(Hs.out)
cat("Expected heterozygosity across overall\n")
print(Hs(nameko.SSR.MLG.overall.genind)[[1]])
cat("\n")

#Baisit diversity
basic.summary <- basic.stats(nameko.SSR.MLG.three.categories.genind)
#cat("\n")
#cat("Basic statistic across overall\n")
#print(basic.summary$overall)
```
