---
title: "GenotypeAccum"
author: "Akira Hirao"
date: "`r Sys.Date()`"
output:
  md_document:
    toc: true
    variant: markdown_github
  html_document:
    toc: true
---

# Genotype accumulation curve

## Loading packages

```{r message = FALSE, warning = FALSE, echo = TRUE}
# Loading packages
library(tidyverse)
library(polysat)
library(hierfstat)
library(adegenet)
library(poppr)
library(RColorBrewer)
```

## SE function

```{r message = FALSE, warning = FALSE, echo = TRUE}
# Defining SE function
std_mean <- function(x) sd(x)/sqrt(length(x))
```

## Loading data set

```{r message = FALSE, warning = FALSE, echo = TRUE}
# Loading data set
nameko.raw <- read_csv("MLG_Pmicro_123samples.csv")

# Comvert to genotype as expressed in 6-digit-numeric code
nameko.6_digit.genotype.raw = tibble::tibble(nameko.raw[,c(1:3)],
  Phmi01 = str_c(formatC(nameko.raw$Phmi01A,width=3, flag="0"),formatC(nameko.raw$Phmi01B,width=3, flag="0")),
  Phmi02 = str_c(formatC(nameko.raw$Phmi02A,width=3, flag="0"),formatC(nameko.raw$Phmi02B,width=3, flag="0")),
  Phmi03 = str_c(formatC(nameko.raw$Phmi03A,width=3, flag="0"),formatC(nameko.raw$Phmi03B,width=3, flag="0")),
  Phmi05 = str_c(formatC(nameko.raw$Phmi05A,width=3, flag="0"),formatC(nameko.raw$Phmi05B,width=3, flag="0")),
  Phmi07 = str_c(formatC(nameko.raw$Phmi07A,width=3, flag="0"),formatC(nameko.raw$Phmi07B,width=3, flag="0")),
  Phmi08 = str_c(formatC(nameko.raw$Phmi08A,width=3, flag="0"),formatC(nameko.raw$Phmi08B,width=3, flag="0")),
  Phmi09 = str_c(formatC(nameko.raw$Phmi09A,width=3, flag="0"),formatC(nameko.raw$Phmi09B,width=3, flag="0")),
  Phmi10 = str_c(formatC(nameko.raw$Phmi10A,width=3, flag="0"),formatC(nameko.raw$Phmi10B,width=3, flag="0")),
  Phmi13 = str_c(formatC(nameko.raw$Phmi13A,width=3, flag="0"),formatC(nameko.raw$Phmi13B,width=3, flag="0")),
  Phmi14 = str_c(formatC(nameko.raw$Phmi14A,width=3, flag="0"),formatC(nameko.raw$Phmi14B,width=3, flag="0")),
  Phmi17 = str_c(formatC(nameko.raw$Phmi17A,width=3, flag="0"),formatC(nameko.raw$Phmi17B,width=3, flag="0")),
  Phmi20 = str_c(formatC(nameko.raw$Phmi20A,width=3, flag="0"),formatC(nameko.raw$Phmi20B,width=3, flag="0")),
  Phmi23 = str_c(formatC(nameko.raw$Phmi23A,width=3, flag="0"),formatC(nameko.raw$Phmi23B,width=3, flag="0")),
  Phmi24 = str_c(formatC(nameko.raw$Phmi24A,width=3, flag="0"),formatC(nameko.raw$Phmi24B,width=3, flag="0"))
)

#filtering out the sample "K23" because of missing alleles
nameko.6_digit.genotype  <- nameko.6_digit.genotype.raw[-122,]

# Convert to genind
nameko.SSR.genind <- df2genind(nameko.6_digit.genotype[,-c(1:4)],ploidy=2,ncode=3,ind.name=nameko.6_digit.genotype$ID,pop=nameko.6_digit.genotype$Pop)

# Convert to genind
nameko.SSR.subpop.genind <- df2genind(nameko.6_digit.genotype[,-c(1:4)],ploidy=2,ncode=3,ind.name=nameko.6_digit.genotype$ID,pop=nameko.6_digit.genotype$Subpop)

strata(nameko.SSR.genind) <- data.frame(nameko.6_digit.genotype[,c(2:3)])


# Convert to genclone
nameko.SSR.genclone <- as.genclone(nameko.SSR.genind)
nameko.SSR.subpop.genclone <- as.genclone(nameko.SSR.subpop.genind)

#Repeat motif of each of the 14 SSR loci
pinfreps <- c(3,2,3,2,2,2,2,3,3,3,2,2,3,2)

#Clone correction
nameko.SSR.MLG.genind <- clonecorrect(nameko.SSR.genind)
nameko.SSR.MLG.subpop.genind <- clonecorrect(nameko.SSR.subpop.genind)
```

## Plotting Genotype accumulation curve

```{r message = FALSE, warning = FALSE, echo = TRUE}
# genotype accumulate curve
gac <- genotype_curve(nameko.SSR.genclone, sample = 1000, quiet = TRUE)
```

## Genetic diversity

```{r message = FALSE, warning = FALSE, echo = TRUE}
poppr(nameko.SSR.genclone)
```

```{r message = FALSE, warning = FALSE, echo = TRUE}
poppr(nameko.SSR.subpop.genclone)
```